### PR TITLE
Additional shortcuts

### LINKED ISSUE #348: Additional keyboard shortcuts
Some additional keyboard shortcuts would improve the usability of the client:

in lobby:
crtl-N: create a new game
crtl-S: spectate selected game
crtl-enter: join selected game
crtl-L: filter games
crtl-D: remove game filter

in-game:
crtl-K: set the counters on an object to a specific number.
(Similar to the crtl-P shortcut, but there could also be a drop down menu to set the color of the counters to be added.)

### PR DESCRIPTION
**Pull request by ZeldaZach**
Fix #348 

This adds some of the shortcuts suggested by the ticket #348. Some of the suggestions aren't do-able right now, and I sorted through them to pick which would make the most sense and be of most use.

Suggestions/improvements are welcomed!

### CODE REVIEW

File: cockatrice/src/gameselector.cpp [hunk 1]
```diff
@@ -197,12 +197,20 @@ void GameSelector::actJoin()
...
     if (createButton)
-        createButton->setText(tr("C&reate"));
-    joinButton->setText(tr("&Join"));
+    {
+        createButton->setText(tr("Create"));
+        createButton->setShortcut(QKeySequence("Ctrl+N"));
+    }
+
+    joinButton->setText(tr("Join"));
+    joinButton->setShortcut(QKeySequence("Ctrl+J"));
```
Comment (poixen) [inline cockatrice/src/gameselector.cpp]: ENTER also does the same.

File: cockatrice/src/tab_deck_editor.cpp [hunk 2]
```diff
@@ -336,7 +336,8 @@ void TabDeckEditor::retranslateUi()
     aLoadDeck->setText(tr("&Load deck..."));
     aSaveDeck->setText(tr("&Save deck"));
     aSaveDeckAs->setText(tr("Save deck &as..."));
-    aLoadDeckFromClipboard->setText(tr("Load deck from cl&ipboard..."));
+    aLoadDeckFromClipboard->setText(tr("Load deck from clipboard..."));
+    aLoadDeckFromClipboard->setShortcut(QKeySequence("Ctrl+L"));
```
Comment (poixen) [inline cockatrice/src/tab_deck_editor.cpp]: Doesn't this collide with the accelerator on line 336.

Comment (Daenyth) [inline cockatrice/src/tab_deck_editor.cpp]: Load deck from clipboard should be ctrl+V, no?

Comment (ZeldaZach) [inline cockatrice/src/tab_deck_editor.cpp]: Using <kbd>Ctrl + V</kbd> combination caused issues on Mac OSX and "L" for "Load" seems more realistic to me. I'm fine changing this back if that's what you'd like.

File: cockatrice/src/gameselector.cpp (hunk 1 above)
Comment (ZeldaZach) [inline cockatrice/src/gameselector.cpp]: Really? Didn't work for me when testing

**Comment by ZeldaZach** [comment]
Any further comments on this?

**Comment by ctrlaltca** [comment]
As @poixen stated, in line 336:
```
aLoadDeck->setText(tr("&Load deck..."));
```
The & before the L will automatically create a Ctrl+L hotkey for "Load Deck".
Then, at line 340, another hotkey is created using the same key sequence>
```
aLoadDeckFromClipboard->setShortcut(QKeySequence("Ctrl+L"));
```
Only one of the hotkeys will work, so i see no point in keeping them both with the same key sequence.

@Daenyth pointed out that currently Ctrl+V is "paste deck from clipboard" and Ctrl+C is "copy deck to clipboard". I suppose Ctrl+V it's quite used since it's the most handy way to get a decklist from a website into cockatrice, and a I'd avoid changing it.

> Using Ctrl + V combination caused issues on Mac OSX

What issues? i just tested it and it seems to work fine.

File: cockatrice/src/tab_deck_editor.cpp (hunk 2 above)
Comment (ZeldaZach) [inline cockatrice/src/tab_deck_editor.cpp]: Reverted

**Comment by ZeldaZach** [comment]
Anything further that needs to be done?

File: cockatrice/src/gameselector.cpp (hunk 1 above)
Comment (poixen) [inline cockatrice/src/gameselector.cpp]: When you have a game selected, press ENTER to join

Comment (ZeldaZach) [inline cockatrice/src/gameselector.cpp]: Doesn't work for me on OS X... weird

Comment (poixen) [inline cockatrice/src/gameselector.cpp]: Try Cmd + O

Comment (ZeldaZach) [inline cockatrice/src/gameselector.cpp]: Yeah, that works @poixen

Comment (poixen) [inline cockatrice/src/gameselector.cpp]: Turns out that is the Mac version of activated. The windows one is return. I would either leave it as Ctrl O or change the Ctrl J to the return key.

File: cockatrice/src/gameselector.cpp [hunk 3]
```diff
@@ -197,12 +197,20 @@ void GameSelector::actJoin()
 void GameSelector::retranslateUi()
 {
     setTitle(tr("Games"));
-    filterButton->setText(tr("&Filter games"));
-    clearFilterButton->setText(tr("C&lear filter"));
+    filterButton->setText(tr("Filter games"));
+    filterButton->setShortcut(QKeySequence("Ctrl+F"));
```
Comment (ctrlaltca) [inline cockatrice/src/gameselector.cpp]: Cmd+F overrides the "fullscreen" action here.

**Comment by ctrlaltca** [comment]
What about adding the actions to the "Room"menu so that the shortcuts are visible to the user somewhere?

**Comment by Daenyth** [comment]
+1 for menu

**Comment by ZeldaZach** [comment]
Closing in favor of #1382

**Pull request closed without merging**